logging/logger_registry.go

```diff
@@ -149,6 +149,16 @@ func LoggerNamed(name string) (logger Logger, ok bool) {
 	return globalLoggerRegistry.loggerNamed(name)
 }
 
+// GetOrNewLogger returns a logger with the specified name if it exists, otherwise it
+// creates and registers one with the same name.
```

Comment:
Could we leave a `TODO` comment to consider removing this with https://viam.atlassian.net/browse/RSDK-8291? See my other comment on why that ticket is relevant.